grouper/fe/handlers/user_shell.py

```diff
@@ -1,21 +1,25 @@
-from grouper.constants import USER_ADMIN, USER_METADATA_SHELL_KEY
+from grouper.constants import USER_METADATA_SHELL_KEY
 from grouper.fe.forms import UserShellForm
 from grouper.fe.settings import settings
 from grouper.fe.util import GrouperHandler
 from grouper.models.audit_log import AuditLog
 from grouper.models.user import User
-from grouper.user_permissions import user_has_permission
+from grouper.service_account import can_manage_service_account
 
 
 class UserShell(GrouperHandler):
+
+    @staticmethod
+    def check_access(session, actor, target):
+        return (actor.name == target.name or
+            (target.role_user and can_manage_service_account(session, actor, tuser=target)))
+
     def get(self, user_id=None, name=None):
         user = User.get(self.session, user_id, name)
         if not user:
             return self.notfound()
 
-        if user.name != self.current_user.name and not (
```

Comment:
Yes. Most other similar user functions also do not allow admins to execute the action, so I changed this to bring it in line with those (i.e. adding tokens, public keys, etc.)